converters/convert.ts

```diff
@@ -0,0 +1,5 @@
+import fromarray from "./fromarray";
+import { discoverToEvents } from "./fromarray/discover";
+import toarray from "./toarray";
+
+export default { fromarray, toarray, discoverToEvents };
```

Comment:
>fromarray [](start = 17, length = 9)

Don't bke the details in this so much. Just think of "in and out" Then you have the converters going in and out. See comment about defining an interface to the conversion to/from.